lms/djangoapps/oauth_dispatch/tests/test_dot_overrides.py

```diff
@@ -0,0 +1,64 @@
+"""
+Test of custom django-oauth-toolkit behavior
+"""
+
+from django.contrib.auth.models import User
+from django.test import TestCase, RequestFactory
+from ..dot_overrides import authenticate, EdxOAuth2Validator
+
+class AuthenticateTestCase(TestCase):
+    """
+    Test that users can authenticate with either username or email
+    """
+
+    def setUp(self):
+        super(AuthenticateTestCase, self).setUp()
+        self.user = User.objects.create_user(
+            username='darkhelmet',
+            password='12345',
+            email='darkhelmet@spaceball_one.org',
+        )
+
+    def test_authenticate_with_username(self):
+        user = authenticate(username='darkhelmet', password='12345')
+        self.assertEqual(
+            self.user,
+            user
+        )
+
+    def test_authenticate_with_email(self):
+        user = authenticate(username='darkhelmet@spaceball_one.org', password='12345')
+        self.assertEqual(
+            self.user,
+            user
+        )
+
+
+class CustomValidationTestCase(TestCase):
+    """
+    Test custom user validation works.
+
+    In particular, inactive users should be able to validate.
+    """
+    def setUp(self):
+        super(CustomValidationTestCase, self).setUp()
+        self.user = User.objects.create_user(
+            username='darkhelmet',
+            password='12345',
+            email='darkhelmet@spaceball_one.org',
+        )
+        self.validator = EdxOAuth2Validator()
+        self.request_factory = RequestFactory()
+
+    def test_new_user_is_active(self):
```

Comment:
This method is actually testing `User.objects.create_user` instead of `EdxOAuth2Validator`.  But I could see the value of verifying your assumptions of the result of `create_user`.  So how about moving this assertion as the first line of the `test_active_user_validates` method?